### PR TITLE
chore: create insert before first void block global plugin

### PR DESCRIPTION
**Pull request by cezarsmpio**
It works for:

- HR
- Embedded asset
- Embedded entry

Any other void block?

### CODE REVIEW

File: packages/rich-text/src/plugins/InsertBeforeFirstVoidBlock/index.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            type: BLOCKS.PARAGRAPH,
+            data: {},
+            children: [{ text: '' }],
+          };
+
+          // It adds two paragraphs outside a timeout and after moving the cursor to the previous line, not sure why...
+          setTimeout(() => {
+            Transforms.insertNodes(editor, paragraph, { at: currentPath as Path });
+            moveToThePreviousLine(editor);
+          }, 0);
```
Comment (cezarsmpio) [inline packages/rich-text/src/plugins/InsertBeforeFirstVoidBlock/index.tsx:31]: Without the timeout:

- Editor -> HR + Paragraph -> Enter **becomes** Editor -> Paragraph + Paragraph (selection) + HR + Paragraph

With the timeout:

- Editor -> HR + Paragraph -> Enter **becomes** Editor -> Paragraph (selection) + HR + Paragraph

Comment (cezarsmpio) [reply packages/rich-text/src/plugins/InsertBeforeFirstVoidBlock/index.tsx:31]: 🤷

File: cypress/integration/RichTextEditor.spec.ts [hunk 2]
```diff
@@ -302,6 +302,25 @@ describe('Rich Text Editor', () => {
         );
         expectRichTextFieldValue(expectedValue);
       });
+
+      // TODO: Cypress throwing after {enter}: Cannot resolve a Slate node from DOM node: [object HTMLSpanElement]
+      // Question: Can we fix it by upgrading the Cypress version? Is this one of those issues that happen because Cypress doesn't fire all the events Slate needs in its current version?
```
Comment (z0al) [inline cypress/integration/RichTextEditor.spec.ts:307]: I'd really like us to upgrade to the latest Cypress sooner than later. We are waaay behind. But that's not something trivial.

**Review by z0al — COMMENTED** [review]
Looks good 👍🏻  but let's try to ask about the timeout issue in Slack? we don't want this to be another source of bugs later 😕

**Comment by z0al** [comment]
@cezarsmpio You may want to take a look at the [trailing-block](https://github.com/udecode/plate/blob/main/packages/trailing-block/src/createTrailingBlockPlugin.ts) plugin as it's very similar to what you're doing but inserts at the very end instead of at the beginning?

**Comment by cezarsmpio** [comment]
@z0al they fix different things. This PR is for empty paragraphs above of first void blocks only.

**Comment by cezarsmpio** [comment]
@z0al I also asked the Plate team about the timeout, you can follow the thread here if you wish: https://slate-js.slack.com/archives/C013QHXSCG1/p1631022429081900

File: packages/rich-text/src/plugins/InsertBeforeFirstVoidBlock/index.tsx (hunk 1 above)
Comment (z0al) [reply packages/rich-text/src/plugins/InsertBeforeFirstVoidBlock/index.tsx:31]: FYI @cezarsmpio I tried the suggestion from Slack (i.e. `Editor.withoutNormalizing`), it doesn't fix this issue.

Comment (cezarsmpio) [reply packages/rich-text/src/plugins/InsertBeforeFirstVoidBlock/index.tsx:31]: Ok, thanks!

File: cypress/integration/RichTextEditor.spec.ts (hunk 2 above)
Comment (cezarsmpio) [reply cypress/integration/RichTextEditor.spec.ts:307]: We've tried to update it but all other tests from other packages started failing.

**Review by z0al — APPROVED** [review]
Approving in case you want to merge it as it's and follow up. But ideally, we want to figure out the `setTimeout` issue or add some working tests before merging this

**Pull request merged**